src/wp-includes/sitemaps/providers/class-wp-sitemaps-posts.php

```diff
@@ -0,0 +1,221 @@
+<?php
+/**
+ * Sitemaps: WP_Sitemaps_Posts class
+ *
+ * Builds the sitemaps for the 'post' object type.
+ *
+ * @package WordPress
+ * @subpackage Sitemaps
+ * @since 5.5.0
+ */
+
+/**
+ * Posts XML sitemap provider.
+ *
+ * @since 5.5.0
+ */
+class WP_Sitemaps_Posts extends WP_Sitemaps_Provider {
+	/**
+	 * WP_Sitemaps_Posts constructor.
+	 *
+	 * @since 5.5.0
+	 */
+	public function __construct() {
+		$this->name        = 'posts';
+		$this->object_type = 'post';
+	}
+
+	/**
+	 * Returns the public post types, which excludes nav_items and similar types.
+	 * Attachments are also excluded. This includes custom post types with public = true.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array Map of registered post type objects (WP_Post_Type) keyed by their name.
+	 */
+	public function get_object_subtypes() {
+		$post_types = get_post_types( array( 'public' => true ), 'objects' );
```

Comment:
`public` implies `rewrite`, but aren't they mutually exclusive?

Reply:
I don't see how `rewrite` is relevant here.